### PR TITLE
Add support for reacting to posts

### LINKED ISSUE #12: Add ability to write reactions
Right now we can only display. Format:

```
{
  type: 'vote',
  vote: {
    link: '%bOMvZd2pEgchYiAbKVALYLz0y2+tpTxxPztYOrFs04A=.sha256',
    value: 1,
    expression: '😄'
  }
}
```

### PR DESCRIPTION
**Pull request by KyleMaas**
Adds the ability to react to posts.  Fixes #12.

The existing logic for dislikes didn't work.  This fixes the bugs in that, but it still seems like a weird way to do things - once someone unlikes a post, no votes of any kind will show for that post from that person again, even if they post another vote later.  Is that how it's supposed to work?

### CODE REVIEW

File: ui/ssb-msg.js [hunk 1]
```diff
@@ -38,12 +38,25 @@ Vue.component('ssb-msg', {
...
         <span v-if="isOOO"><a href="javascript:void(0);" v-on:click="getOOO">get msg</a></span>
+        <div class='reactions'>
+	  <span class='reactions-existing'>
+            <span v-for="reaction in reactions">
+              <span v-bind:title="reaction.author">{{ reaction.expression }}</span>
+            </span>
+	  </span>
+	  <span class='reactions-mine' v-if="myReactions.length > 0">
+	    <span v-for="reaction in myReactions">
+	      <a href="javascript:void(0);" v-on:click="unlike()">{{ reaction.expression }}</a>
```
Comment (arj03) [inline ui/ssb-msg.js:50]: A:  title="remove reaction" would be nice here

File: ui/ssb-msg.js [hunk 2]
```diff
@@ -88,6 +104,31 @@ Vue.component('ssb-msg', {
         else
           this.msg = { key: this.msg.key, value: msgValue }
       })
+    },
+    react: function(emoji) {
+      if(emoji == 'Unlike') {
+	this.myReactions = [];
```
Comment (arj03) [inline ui/ssb-msg.js:110]: can you check indention here and remove the ; ? So it conforms with the rest of the code :-)

File: ui/ssb-msg.js [hunk 3]
```diff
@@ -88,6 +104,31 @@ Vue.component('ssb-msg', {
...
+      } else {
+        this.myReactions.push({ expression: emoji });
+      }
+      var reactTo = this.msg.key;
+      var message = {
+        type: 'vote',
+	vote: {
```
Comment (arj03) [inline ui/ssb-msg.js:117]: ditto

File: ui/ssb-msg.js [hunk 4]
```diff
@@ -88,6 +104,31 @@ Vue.component('ssb-msg', {
...
+	  value: 1,
+	  expression: emoji
+	}
+      };
+      SSB.db.publish(message, (err) => {
+        if (err) console.log(err)
+      })
+    },
+    unlike: function() {
+      if(confirm("Are you sure you want to unlike this post?"))
```
Comment (arj03) [inline ui/ssb-msg.js:128]: remove reaction instead of unlike

File: ui/ssb-msg.js [hunk 5]
```diff
@@ -88,6 +104,31 @@ Vue.component('ssb-msg', {
...
+	this.myReactions = [];
+      } else {
+        this.myReactions.push({ expression: emoji });
+      }
+      var reactTo = this.msg.key;
+      var message = {
+        type: 'vote',
+	vote: {
+	  link: reactTo,
+	  value: 1,
```
Comment (arj03) [inline ui/ssb-msg.js:119]: For unlike the value needs to be [0](https://github.com/ssbc/patchwork/blob/ae29bd7f91e16deac4bd88b099175fa85a419815/lib/depject/message/html/actions.js#L58)

File: ui/ssb-msg.js [hunk 6]
```diff
@@ -112,24 +153,30 @@ Vue.component('ssb-msg', {
     SSB.db.query(
       and(votesFor(this.msg.key)),
       toCallback((err, msgs) => {    
-        const unlikes = msgs.filter(x => x.value.content.vote.expression == 'Unlike').map(x => { author: x.value.author })
-        this.reactions = msgs.map(x => {
+        if(err) {
+	  console.log("Error getting votes: " + err);
+	  return;
+	}
+        const unlikes = msgs.filter(x => x.value.content.vote.expression == 'Unlike').map(x => { return x.value.author })
```
Comment (arj03) [inline ui/ssb-msg.js:160]: better to check for 0 here instead of unlike

**Comment by arj03** [comment]
This is really fantastic! 

I added a few minor review suggestions.

**Comment by KyleMaas** [comment]
How's that?  I think my vim indentation settings were set for tabs.

**Comment by arj03** [comment]
Excellent! Thanks a lot.

**Pull request merged**